pkg/storage/tsdb/caching_config.go

```diff
@@ -105,7 +106,7 @@ func CreateCachingBucket(chunksConfig ChunksCacheConfig, metadataConfig Metadata
 	}
```

Comment:
Above, I think you need to add the `thanos_` prefix to the registry passed to `cache.CreateClient("chunks-cache")` and `cache.CreateClient("metadata-cache")`. If I'm right, then I'm suprised no test failed. Is there any existing test where we can improve assertions?

Pay attention that below `reg` is also used, so just replacing `reg` with a wrapped one is not correct, but we just need to wrap the `reg` passed to `cache.CreateClient("chunks-cache")` and `cache.CreateClient("metadata-cache")`.

Reply:
Added missing `thanos_` metric prefix for chunks and metadata caches, and improved integration test assertions. 

done @ 3226c2c9c949ac10d764d0f93b27788c932d7f69